layers/core_checks/cc_render_pass.cpp

```diff
@@ -5104,3 +5113,226 @@ bool CoreChecks::ValidateInheritanceInfoFramebuffer(VkCommandBuffer primaryBuffe
+                const Location loc =
+                    loc_info.dot(Struct::VkRenderingAttachmentLocationInfoKHR, Field::pColorAttachmentLocations, i);
+                skip |= LogError("VUID-VkRenderingAttachmentLocationInfoKHR-pColorAttachmentLocations-09513", objlist, loc,
+                                 "= %" PRIu32 " have same value as pColorAttachmentLocations[%" PRIu32 "] = %" PRIu32, location,
+                                 unique[location], location);
+            } else
+                unique[location] = i;
+        }
+    }
+
+    if (pLocationInfo->colorAttachmentCount > phys_dev_props.limits.maxColorAttachments) {
+        skip |= LogError("VUID-VkRenderingAttachmentLocationInfoKHR-colorAttachmentCount-09514", objlist,
+                         loc_info.dot(Field::colorAttachmentCount),
+                         "(%" PRIu32 ") is greater than the maxColorAttachments limit (%" PRIu32 ").",
+                         pLocationInfo->colorAttachmentCount, phys_dev_props.limits.maxColorAttachments);
+    }
+
+    return skip;
+}
+
+bool CoreChecks::PreCallValidateCmdSetRenderingAttachmentLocationsKHR(VkCommandBuffer commandBuffer,
+                                                                      const VkRenderingAttachmentLocationInfoKHR *pLocationInfo,
+                                                                      const ErrorObject &error_obj) const {
+    const auto &cb_state = *GetRead<vvl::CommandBuffer>(commandBuffer);
+    bool skip = false;
+
+    if (!enabled_features.dynamicRenderingLocalRead) {
+        skip |= LogError("VUID-vkCmdSetRenderingAttachmentLocationsKHR-dynamicRenderingLocalRead-09509", commandBuffer,
+                         error_obj.location, "dynamicRenderingLocalRead was not enabled.");
+    }
+
+    skip |= ValidateCmd(cb_state, error_obj.location);
+
+    const auto *rp_state_ptr = cb_state.activeRenderPass.get();
+    if (!rp_state_ptr) {
+        return skip;
+    }
+
+    const auto &rp_state = *rp_state_ptr;
+
+    if (!rp_state.UsesDynamicRendering()) {
+        const LogObjectList objlist(commandBuffer, rp_state.VkHandle());
+        skip |= LogError("VUID-vkCmdSetRenderingAttachmentLocationsKHR-commandBuffer-09511", objlist, error_obj.location,
+                         "Please use vkCmdBeginRendering.");
+    }
+
+    if (pLocationInfo && pLocationInfo->colorAttachmentCount != rp_state.dynamic_rendering_begin_rendering_info.colorAttachmentCount) {
+        const LogObjectList objlist(commandBuffer, rp_state.VkHandle());
+        skip |= LogError("VUID-vkCmdSetRenderingAttachmentLocationsKHR-pLocationInfo-09510", objlist, error_obj.location,
+                         "colorAttachmentCount (%" PRIu32 ") is not equal to count specified in VkRenderingInfo (%" PRIu32 ").",
+                         pLocationInfo->colorAttachmentCount, rp_state.createInfo.attachmentCount);
+    }
+
+    skip |= ValidateRenderingAttachmentLocationsKHR(pLocationInfo, commandBuffer, error_obj.location);
```

Comment:
near the top of the function go

```cpp
const Location loc_info = error_obj.location.dot(Field::pLocationInfo);
```

then here

```suggestion
    skip |= ValidateRenderingAttachmentLocationsKHR(pLocationInfo, commandBuffer, info_loc);
```

Reply:
> near the top of the function go

Will add.